main.py

```diff
@@ -121,8 +162,69 @@ def write_candidate(perm: Permuter, source: str) -> None:
             break
         except FileExistsError:
             pass
-    print(f"wrote to {fname}")
+    print(f"wrote to {fname}")   
+    
+def post_score(context: EvalContext, permuter: Permuter, cand: Candidate, exception: BaseException, profiler: Profiler) -> None:
+    if exception != None:
+        print(f"[{permuter.unique_name}] internal permuter failure.")
+        traceback.print_exc()
+        print(f"To reproduce the failure, rerun with: --seed {permuter.seed}")
+        exit(1)
+
+    if options.print_diffs:
+        permuter.print_diff(cand)
+        yn = input("Press any key to continue...")
+
+    context.iteration += 1
+    if cand.score_value is None:
+        context.errors += 1
+    disp_score = 'inf' if cand.score_value == permuter.scorer.PENALTY_INF else cand.score_value
+    timings = ''
+    if options.show_timings:
+        assert(profiler)
```

Comment:
`profiler` is not Optional, so this assert doesn't seem meaningful assuming the program type-checks

(also, remove parentheses, assert is a keyword)